### DISCUSSION TITLE
Insertion de commentaires

### DISCUSSION THREAD

**Discussion by Hant65** [Jean-Romain/tabliato#30] [General]
Bonjour, 

D'abord un grand merci pour l'initiative tabliato et sa diffusion ! Elle me permet de mettre au propre les notations sauvages de mes débuts, et de retrouver ainsi des airs oubliés depuis bien longtemps ! 

Ensuite, une question : j'aurais aimé pouvoir insérer un ou plusieurs commentaires à l'intérieur d'une page, par exemple entre deux ensembles "portée/tablature" pour indiquer que ce qui suit va être une variation, ou en bas de page pour indiquer la succession des parties (A, B, C) d'un morceau lorsque les reprises sont un peu complexes (style AA B CC A B). 

J'ai essayé de le faire avec la commande `\markup { }` de `Lilypond`, mais  cela ne fonctionne pas. Y a t il un autre moyen ? 

Bien cordialement

**Comment by Jean-Romain** [comment]
Bonjour, aucune idée désolé. Est ce que je vous pourriez produire une image pour me montrer à quoi vous voulez que ca ressemble ?

**Reply by Hant65** [reply]
Bonsoir, 
 Voici l'image (enfin, j'espère l'avoir insérée comme il faut !). L'idée serait de remplacer les mentions manuscrites par du texte à saisir dans le code sous Tabliato. Sous Lilypond, cela semble possible via l'instruction \markup (cf. § 3.1.5 de la doc Lilypond). Mais Tabliato ne semble pas comprendre cette instruction (à moins que je ne sache pas l'utiliser) !
  ![IMG_20230128_202932(1)](https://user-images.githubusercontent.com/122602106/215287818-00b50876-2191-4c25-b3ed-1fa7ddeaee8d.jpg)

**Reply by Jean-Romain** [reply]
Je vois. Je vais chercher voir ce qu'on peut faire. De mon côté je mettais les variantes dans une deuxième partitions donc c'est une fonctionnalité utile. Je pense que c'est plus compliqué à faire que ca en à l'air.

**Comment by Jean-Romain** [comment]
Vous pouvez utiliser la commande `\mark`

```ly
\mark "Partie 1"
```

On peut aussi customiser le rendu. J'aime bien celui ci

```ly
\mark \markup{ \override #'(box-padding . 0.5) \box  \caps "Partie 1"}
```

Un example complet

![Capture d’écran du 2023-02-02 20-49-27](https://user-images.githubusercontent.com/3872279/216492644-20b85057-b89c-4a98-bbfe-ab462c544dbc.png)



```ly
\mark \markup{ \override #'(box-padding . 0.5) \box  \caps "Partie 1"}

\repeat volta 2 
{
  t5:2.
  4'
  5
  5:4 4' 5	
  p5:2.
  4'
  5
  5:4 4' 5

   A A A A G G G G 
  
  \break
  
  t4:2.
  3'
  4
  4:4 3' 4
  p4:2.
  3'
  4
  4:4 3' 5 

   A A A A G G G G
}

\break

\mark \markup{ \override #'(box-padding . 0.5) \box  \caps "Partie 2"}

\repeat volta 2
{
  t5:2. 4' 7
  5:4 7 5		
  p5:2. 4' 7
  5:4 7 5

   A A A A G G G G 
  
  \break
  
  t4:2. 5 8
  8:4 6' 7
  p7:2. 6' 7
  5:4 4' 5

 A A A A G G G G
}
```

**Reply by Hant65** [reply]
Merci ! Cela résout la question ;  dans le manuel Lilypond, je n'avais donc pas su voir cette instruction \mark. 
Bien cordialement,

**Reply by Jean-Romain** [reply]
Pour trouver ce qu'on veut dans lilypond il faut une bonne expérience. Le logiciel est tellement riche qu'il est vraiment compliqué. En plus l'idée du `markup` était sûrement correcte mais ca n'a pas marché car les markup sont désactivés sur la porté dans le fichier produit par `tabliato` et ca, vous ne pouviez pas le savoir.

Merci pour la question. J'avais besoin de trouver une solution moi aussi depuis longtemps.

**Reply by Jean-Romain** [reply]
Vous pouvez ajouter `\tweak X-offset #0` avant le `\mark` pour l'alignement. Dans la prochaine version j'ajouterai un commande automatique pour faire tout ca.

**Reply by Hant65** [reply]
Merci pour ce conseil supplémentaire ! Effectivement cela fait plus propre !! 
Je ne voudrais pas abuser... mais j'ai une autre question : dans Tabliato, est il possible de changer le tempo au cours d'un morceau (cas de quelques scottish / valse où l'on passe d'un 2/4 à un 3/4). Je pense que non, vu que le tempo est déclaré dans le panneau de gauche et s'applique donc a priori à l'ensemble du morceau. Mais j'ai peut-être zappé une option ? Ceci dit, ce genre de morceau n'est pas très fréquent, et si cela doit conduire à révolutionner Tabliato, cela n'en vaut sans doute pas la peine. Encore merci !

**Reply by Jean-Romain** [reply]
Oui c'est possible. Je pense même qu'il y a un exemple dans les tablatures du site mais je ne le trouve pas (**edit:** trouvé c'est [Cotillons verts](https://jean-romain.github.io/tabliato/tablatures/Cotillons%20verts/Cotillons%20verts.pdf))

En fait l'interface graphique rend plus facile d’accès certaines options mais c'est le code qui compte. On peut réécrire  le rythme avec `\time` et le motif avec `\motif`. Mais ce n'est pas documenté vous ne pouviez donc pas le deviner.

Ca donne ca

```
p7:4 t6 p6 E
t7 6':4. p5':8 D
t6:4  p4':2 E
r:8 p4:8 4' t6 p6 7  E
t9:4 6' 8 A
7' 9:4. 8':8 D
p7':4 7 6 E 
p4:2 \tuplet 3/2 { 4:8 4' 6 } E

\break

\time 7/8
\motif [B:4. a:4 a:4]

t8:8 6' 7 8 6' 7 6':16 8 A
7':4 7:8 7':4 7' D
7':8 9 8' 9 7' 8  7' G
p7':4 t8':8 p7':4 7' C
```
![Capture d’écran du 2023-02-06 15-35-09](https://user-images.githubusercontent.com/3872279/217079306-291392e0-486f-4177-ae74-1fc74b3e940a.png)

**Reply by Jean-Romain** [reply]
> Je ne voudrais pas abuser... 

Pas de trouble, je suis bien content que ce logiciel serve à quelqu'un d'autre que moi :wink:  Je vous demanderais juste d'ouvrir une nouvelle discussion pour chaque question histoire que ce soit rangé et que d'autres aient une chance de retrouver les infos dans le futur (sait on jamais)

Et si vous voulez faire un petit geste en remerciement pour ce logiciel gratuit, vous pourriez partager vos tablatures (le fichier tabliato `.dtb`) pour que je les ajoute sur le site. Ca serait bien apprécié :+1:

**Reply by Hant65** [reply]
C'est noté, à la fois pour le rangement des discussions, et le partage de tablatures qui viendront sous peu (un peu de tri à faire). 
Encore merci !

**Reply by Hant65** [reply]
Suite à nos échanges, voici une petite contribution pour la ... TabliatoThèque. Je continuerai à l'occasion.  Encore merci. 
[PourTabliato.zip](https://github.com/Jean-Romain/tabliato/files/10684895/PourTabliato.zip)

**Reply by Jean-Romain** [reply]
Super merci. Je vois que la moitié des partitions sont écrites pour le clavier nommé "monaccordéon". N'ayant pas ce fichier de clavier je ne peux pas gérérer le rendu. Est ce que c'est un accordéon spécial ? Peut on écrire la tablature pour un accordéon classique ?

**Reply by Hant65** [reply]
Oh, je vois. "monaccordéon" est un Honher 2 rangs 8 basses G/C des années 80 du siècle dernier, avec la particularité d'un Sol dièse (celui qui suit le Sol du bouton p6) en poussant sur le bouton 1'. Je crois que c'était d'origine, à l'époque. Je n'ai pas retrouvé Sol dièse en poussé parmi les claviers intégrés à Tabliato, et donc j'ai créé le fichier qui me va bien, et je le laisse par défaut (parce que c'est avec lui que je joue). Ceci dit, ce clavier n'est nécessaire que pour les morceaux utilisant ce Sol dièse : uniquement l'air portugais dans l'envoi que je vous ai fait ; Tous les autres peuvent être joués avec le "Viennois 2 rangs 8 basses sol do". Qu'est ce qui pourrait vous arranger : je vous envoie le fichier de mon clavier ? ou la partition et le .mid de cet air portugais ? ou on oublie cet air portugais, et je régénère les différents .dtb en déclarant le viennois ?

**Reply by Jean-Romain** [reply]
Merci de votre réponse. Je vais régénérer les autres avec un clavier standard. Je pourrais ajouter votre clavier par defaut si vous le désirez.

**Reply by Jean-Romain** [reply]
Petit conseil: 

plutôt que 

```r
\motif [B:4 a:4]
```

je vous conseille d’écrire

```r
\motif [B:8 r:8 a:8 r:8]
```

Visuellement ca ne change rien, mais le rendu audio est plus dynamique et ca correspond mieux à ce qu'on joue en vrai. C'est ce que produit tabliato quand on donne un rythme dans la liste.

**Reply by Hant65** [reply]
Je joins à ce message le fichier .csv de mon vieil Hohner.  Comme je vous disais, je ne me rappelle pas avoir fait changer sur cet accordéon la note produite en poussant ce fameux bouton 1' , et si c'est bien le cas, cela peut être utile de l'intégrer dans Tabliato ; à voir comment le nommer pour que les utilisateurs comprennent de quoi il s'agit. Je me souviens l'époque, dans notre groupe d'apprentis diatonistes, tous les claviers n'étaient pas identiques en ce qui concerne ce bouton 1' (qu'on n'utilise quant même pas très souvent) ; je ne peux quand même pas garantir que tous les Hohner étaient équipés comme le mien... 
Merci pour les indications \motif. Ça rejoint une de mes préoccupations, je ne maitrise pas encore l'écriture de l'accompagnement, en particulier sur les mesures incomplètes. Mais j'ouvrirai là dessus un nouveau sujet, on est déjà bien loin des commentaires sur la partition ! 
[Hohner_GC_SolDièse1PrimePoussé.csv](https://github.com/Jean-Romain/tabliato/files/10701006/Hohner_GC_SolDiese1PrimePousse.csv)
